google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ReadWriteTransaction.java

```diff
@@ -80,6 +81,7 @@ class ReadWriteTransaction extends AbstractMultiUseTransaction {
   private static final String MAX_INTERNAL_RETRIES_EXCEEDED =
       "Internal transaction retry maximum exceeded";
   private static final int MAX_INTERNAL_RETRIES = 50;
+  private final ReentrantLock abortedLock = new ReentrantLock();
```

Comment:
This is changed to use a `ReentrantLock` instead of a synchronized block, because Java virtual threads cannot be unmounted when they are inside a synchronized block.